src/chunk.c

```diff
@@ -1123,6 +1153,62 @@ chunk_scan_ctx_get_chunk(ChunkScanCtx *ctx)
 	return ctx->data;
 }
 
+/*
+ * Resurrect a chunk from a tombstone.
+ *
+ * A chunk can be dropped while retaining its metadata as a tombstone. Such a
+ * chunk is marked with dropped=true.
+ *
+ * This function resurrects such a dropped chunk based on the original metadata,
+ * including recreating the table and related objects.
+ */
+static Chunk *
+chunk_resurrect(Hypertable *ht, ChunkStub *stub)
+{
+	ScanIterator iterator;
+	Chunk *chunk = NULL;
+	int count = 0;
+
+	Assert(NULL != stub->constraints);
+	Assert(NULL != stub->cube);
+
+	iterator = ts_scan_iterator_create(CHUNK, RowExclusiveLock, CurrentMemoryContext);
+	init_scan_by_chunk_id(&iterator, stub->id);
+
+	ts_scanner_foreach(&iterator)
+	{
+		TupleInfo *ti = ts_scan_iterator_tuple_info(&iterator);
+		HeapTuple new_tuple;
+
+		Assert(count == 0 && chunk == NULL);
+		chunk = palloc0(sizeof(Chunk));
+		chunk_formdata_fill(&chunk->fd, ti->tuple, ti->desc);
+		chunk->fd.dropped = false;
+		new_tuple = chunk_formdata_make_tuple(&chunk->fd, ti->desc);
+		ts_catalog_update_tid(ti->scanrel, &ti->tuple->t_self, new_tuple);
+		heap_freetuple(new_tuple);
+
+		/* Fill in the stub information and create data table and related
+		 * objects while we hold lock on the metadata row */
+		chunk->constraints = stub->constraints;
+		chunk->cube = stub->cube;
+		chunk_create_table_after_lock(ht, chunk);
```

Comment:
where is this functionality used - i.e. when do we need to recreate the table versus just using the metadata for a tombstoned chunk?

Reply:
There are two cases when we create tables: 1. When the chunk is first created (no tombstone) and 2. when it is resurrected (tombstone exists).